feature/mpls/sr/otg_tests/srte_egress_node_forwarding/egress_node_forwading_test.go

```diff
@@ -0,0 +1,674 @@
+
+	return otgConfig
+}
+
+func enableCapture(t *testing.T, config gosnappi.Config, port string) {
+	config.Captures().Clear()
+	t.Log("Enabling capture on ", port)
+	config.Captures().Add().SetName(port).SetPortNames([]string{port}).SetFormat(gosnappi.CaptureFormat.PCAP)
+
+}
+
+func startCapture(t *testing.T, otg *otg.OTG) gosnappi.ControlState {
+	t.Helper()
+	cs := gosnappi.NewControlState()
+	cs.Port().Capture().SetState(gosnappi.StatePortCaptureState.START)
+	otg.SetControlState(t, cs)
+
+	return cs
+}
+
+func stopCapture(t *testing.T, otg *otg.OTG, cs gosnappi.ControlState) {
+	t.Helper()
+	cs.Port().Capture().SetState(gosnappi.StatePortCaptureState.STOP)
+	otg.SetControlState(t, cs)
+}
+
+func processCapture(t *testing.T, otg *otg.OTG, port string) string {
+	bytes := otg.GetCapture(t, gosnappi.NewCaptureRequest().SetPortName(port))
+	time.Sleep(30 * time.Second)
+	capture, err := os.CreateTemp("", "pcap")
+	if err != nil {
+		t.Errorf("ERROR: Could not create temporary pcap file: %v\n", err)
+	}
+	if _, err := capture.Write(bytes); err != nil {
+		t.Errorf("ERROR: Could not write bytes to pcap file: %v\n", err)
+	}
+	defer capture.Close()
+
+	return capture.Name()
+}
+
+func validatePackets(t *testing.T, filename string, protocolType string) error {
+
+	mplsPacketCount := int32(0)
+
+	handle, err := pcap.OpenOffline(filename)
+	if err != nil {
+		return err
+	}
+	defer handle.Close()
+
+	packetSource := gopacket.NewPacketSource(handle, handle.LinkType())
+
+	for packet := range packetSource.Packets() {
+		if protocolType == "ipv4" {
+			if ipLayer := packet.Layer(layers.LayerTypeIPv4); ipLayer != nil {
+				ip, _ := ipLayer.(*layers.IPv4)
+				if ip.DstIP.Equal(net.ParseIP(ateTrafficDstIPv4)) {
+					if mplsLayer := packet.Layer(layers.LayerTypeMPLS); mplsLayer != nil {
+						mplsPacketCount += 1
+					}
+				}
+				if ip.TTL == mplsOuterTTL-1 {
+					t.Logf("TTL decremented by 1")
+					break
+				} else {
+					return fmt.Errorf("Didin't get TTL as expected. Expected: %v, Got: %v", mplsOuterTTL-1, ip.TTL)
+				}
+			}
+		} else {
+			if ipLayer := packet.Layer(layers.LayerTypeIPv6); ipLayer != nil {
+				ip, _ := ipLayer.(*layers.IPv6)
+				if ip.DstIP.Equal(net.ParseIP(ateTrafficDstIPv6)) {
+					if mplsLayer := packet.Layer(layers.LayerTypeMPLS); mplsLayer != nil {
+						mplsPacketCount += 1
+					}
+				}
+				if ip.HopLimit == mplsOuterTTL-1 {
+					t.Logf("TTL decremented by 1")
+					break
+				} else {
+					return fmt.Errorf("Didin't get TTL as expected. Expected: %v, Got: %v", mplsOuterTTL-1, ip.HopLimit)
+				}
+			}
+		}
+	}
+
+	if mplsPacketCount != 0 {
+		return fmt.Errorf("MPLS label is not popped by the DUT")
+	}
+
+	return nil
+}
+
+// verifyTrafficValidations runs traffic and validates ATE flow metrics and DUT counters.
+func verifyTrafficValidations(t *testing.T, dut *ondatra.DUTDevice, ate *ondatra.ATEDevice, flowName string) (totalTxFromATE, totalRxAtATE uint64, err []error) {
```

Comment:
Its more common to return a single error so the caller does not have to do special handling for the returns.

Additional edit suggested below to consolidate `[]error` -> `error`.

```suggestion
func verifyTrafficValidations(t *testing.T, dut *ondatra.DUTDevice, ate *ondatra.ATEDevice, flowName string) (totalTxFromATE, totalRxAtATE uint64, err error) {
```